core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala

```diff
@@ -70,7 +80,9 @@ private[spark] object TaskLocation {
   def apply(str: String): TaskLocation = {
     val hstr = str.stripPrefix(inMemoryLocationTag)
     if (hstr.equals(str)) {
-      if (str.startsWith(executorLocationTag)) {
+      if (str == "*") {
+        WildcardLocation
```

Comment:
If this goes in I would like to see documentation about this being DeveloperAPI and a temporary workaround and will be removed once locality is fixed.